### ISSUE TITLE
CI status on MR page stay grey (and stop spinning0 after a 500 code status is returned

### ISSUE THREAD

**Issue by allan-simon** [gitlabhq/gitlabhq#9070]
I'm using gitlab 7.9.1 and gitlab-ci 5.4.2 (on 2 different instances on 2 machine in the same LAN)  and from times to times the gitlab-ci status icon in the MR page  stop spinning , checking the network panel of firefox show it return a 500 (without anything specific in the  body of the said response) 

I haven't found a way yet to trigger this bug in a repeated way.

**Comment by allan-simon** [comment]
but I think the code should handle that case and put the "error while connecting to the server" message

**Comment by allan-simon** [comment]
upgrading gitlab-ci to 7.9.1 does not solve the issue.

**Comment by allan-simon** [comment]
it seems it's because of a (Chinese) DNS problem 


```
SocketError (getaddrinfo: Temporary failure in name resolution):
app/models/project_services/gitlab_ci_service.rb:49:in `get_ci_build'
app/models/project_services/gitlab_ci_service.rb:53:in `commit_status'
app/controllers/projects/merge_requests_controller.rb:163:in `ci_status'
```

**Comment by Razer6** [comment]
Fixed with GitLab 8.0 , in which CI and GitLab are a single app.
